src/bidiMapper/domains/network/NetworkProcessor.ts

```diff
@@ -68,7 +68,11 @@ export class NetworkProcessor {
 
     const intercept: Network.Intercept = this.#networkStorage.addIntercept({
       urlPatterns: parsedUrlPatterns,
-      phases: params.phases,
+      // We checked that phases has at least one entry
```

Comment:
I don't think this is needed. There's a `params.phases.length === 0` exception right at the beginning. Revert this chunk.

Reply:
TS is not smart enough to understand this sadly.